src/main/java/com/aws/greengrass/componentmanager/KernelConfigResolver.java

```diff
@@ -118,6 +119,10 @@ public KernelConfigResolver(ComponentStore componentStore, Kernel kernel, Nucleu
                 .put(DECOMPRESSED_PATH_KEY, (id) -> nucleusPaths.unarchiveArtifactPath(id).toAbsolutePath().toString());
         systemParameters.put(ARTIFACTS_NAMESPACE, artifactNamespace);
 
+        HashMap<String, CrashableFunction<ComponentIdentifier, String, IOException>> workNamespace = new HashMap<>();
+        workNamespace.put(PATH_KEY, (id) -> nucleusPaths.workPath(id.getName()).toAbsolutePath().toString());
```

Comment:
use `getServiceName` not `getName` otherwise this will be wrong for lambdas

Reply:
You mean `GreengrassService#getServiceName`? Looks like here we only have `ComponentIdentifier`

Reply:
Yes that's what I mean. The service name will be the same as the component ID, so that's fine then.